### PR TITLE
Local regex

### PR DESCRIPTION
**Pull request by 70ray**
make validCharRegex when needed and use native javascript to make validCharRegex since unicode regex now in supported browsers. So don't need $(document).ready() in character_test.js which causes a problem in unit tests if another test uses promises.

### CODE REVIEW

**Comment by cpeel** [comment]
Please throw up a sandbox we can sanity check this in.

Thanks for the additional comments about what JS functions work in what browsers!

File: styles/themes/charcoal.css [hunk 1]
```diff
@@ -146,7 +146,7 @@
   font-style: normal;
   font-weight: 700;
   font-display: swap;
-  src: url(https://fonts.gstatic.com/s/ebgaramond/v19/~ChoKC0VCIEdhcmFtb25kOgsI9NCduwcVAAAvRCAA.ttf) format('truetype');
+  src: url(https://fonts.gstatic.com/s/ebgaramond/v24/SlGDmQSNjdsmc35JDF1K5E55YMjF_7DPuGi-DPNUAw.ttf) format('truetype');
```
Comment (chrismiceli) [inline styles/themes/charcoal.css:149]: Off topic of the PR, not sure we're GDPR compliant, but there is that recent news about leaking ip addresses to google violating data policies
https://termageddon.com/google-fonts-violates-gdpr/

File: scripts/character_test.js [hunk 2]
```diff
@@ -1,37 +1,22 @@
...
+/*global validCharacterPattern XRegExp */
 /* exported testText */
 
-// regex unicode property escape is supported in Chrome and Safari (Feb 2020)
-// but not in Firefox or Edge. Use 3rd party http://xregexp.com/ instead
+// regex unicode property escape is supported in Chrome 64, Safari 11.1
+// Firefox 78, Edge 79, Opera 51. Use 3rd party http://xregexp.com/ instead
 
 // this matches any character: non-mark codepoint followed by 0 or more marks
 const charMatch = XRegExp("\\PM\\pM*", "Ag");
```
Comment (chrismiceli) [inline scripts/character_test.js:8]: are we aiming to get rid of all instances of XRegExp eventually, right?

**Review by chrismiceli — APPROVED** [review]

File: styles/themes/charcoal.css (hunk 1 above)
Comment (cpeel) [reply styles/themes/charcoal.css:149]: Yeah, saw that. pgdp.net is not GDPR compliant with cookies either.

Comment (srjfoo) [reply styles/themes/charcoal.css:149]: I remember this issue coming up once before, but I don't remember where or when. I thought that the answer was that for \<reasons\>, it didn't apply to us, but I can't quote those reasons, and don't know whether the assessment might change over time.

**Comment by srjfoo** [comment]
Regarding `tools/proofers/process_diacritcal_markup.js`, I know it's been like that since the file was created, but would it be a good idea to fix the misspelling? I know the code doesn't care, as long as the references in the code match the filename being referenced, but I went looking everywhere for references to `process_diacritical_markup.js` and of course couldn't find it, and it took a while to realize where I was going wrong.

File: scripts/character_test.js (hunk 2 above)
Comment (70ray) [reply scripts/character_test.js:8]: Yes.

**Comment by 70ray** [comment]
> Regarding `tools/proofers/process_diacritcal_markup.js`, I know it's been like that since the file was created, but would it be a good idea to fix the misspelling?

Yes, we should do that.

**Comment by 70ray** [comment]
> Please throw up a sandbox we can sanity check this in.

Sorry, done this now:
Sandbox at: https://www.pgdp.org/~rp31/c.branch/local_regex

**Comment by 70ray** [comment]
Updated sandbox after last commit. This is is for quiz.

**Review by cpeel — APPROVED** [review]
This LGTM.

**Pull request merged**